kiwi/utils/rpm_database.py

```diff
@@ -81,6 +81,17 @@ def set_macro_from_string(self, setting):
     def write_config(self):
         self.rpmdb_image.write_config()
 
+    def import_signing_key_to_image(self, key):
+        """
+        Import the given signing key to the image rpm database
+        """
+        Command.run(
+            [
+                'rpm', '--root', self.root_dir, '--import', key,
+                '--dbpath', self.rpmdb_host.expand_query('%_dbpath')
```

Comment:
You need the _image_'s dbpath, not the host's.

Reply:
This information is not present at the time the signing key gets imported. The process is as follows:

1. there are signing keys, import them to the image root using the host rpm. At this point the image root is an almost empty directory. Thus this causes the init of a new rpm database which would be normally done at the first invocation of a rpm based package manager.

2. bootstrap install, rpm now has the signing keys available, the install of rpm from the distro could now introduce a new location of the rpm database. This is the first time I would know about that location, not before

3. switch to the new root, doing this involves the move/rebuild of the rpm database to be correct for the distribution we build

4. continue the rest of the install as chroot operation

I have tested this processing as mentioned with all major rpm based distros. So far this concept works nicely

Reply:
Okay. 👍